command_line/symmetry.py

```diff
@@ -64,87 +95,147 @@
-            expt.crystal.set_space_group(sgtbx.space_group("P 1"))
-            expt.crystal = expt.crystal.change_basis(cb_op_inp_best)
-            expt.crystal.set_space_group(best_space_group)
-
-            S = parameter_reduction.symmetrize_reduce_enlarge(
-                expt.crystal.get_space_group()
-            )
-            S.set_orientation(expt.crystal.get_B())
-            S.symmetrize()
-            expt.crystal.set_B(S.orientation.reciprocal_matrix())
-            reindexed_refl = copy.deepcopy(reflections[i])
-            reindexed_refl["miller_index"] = cb_op_inp_best.apply(
-                reindexed_refl["miller_index"]
+        # Reindex the input data
+        experiments, reflection_tables = _reindex_experiments_reflections(
+            experiments, reflection_tables, best_space_group, cb_op_inp_best
+        )
+
+    elif params.laue_group is not None:
+        if params.change_of_basis_op is not None:
+            cb_op = sgtbx.change_of_basis_op(params.change_of_basis_op)
+        else:
+            cb_op = sgtbx.change_of_basis_op()
+        # Reindex the input data
+        experiments, reflection_tables = _reindex_experiments_reflections(
+            experiments, reflection_tables, params.laue_group.group(), cb_op
+        )
+
+    if params.systematic_absences.check:
+        logger.info("=" * 80)
+        logger.info("")
+        logger.info("Analysing systematic absences")
+        logger.info("")
+
+        if (params.d_min is Auto) and (result is not None):
+            d_min = result.intensities.resolution_range()[1]
+        elif params.d_min is Auto:
+            d_min = resolution_filter_from_reflections_experiments(
+                reflection_tables,
+                experiments,
+                params.min_i_mean_over_sigma_mean,
+                params.min_cc_half,
             )
-            reindexed_reflections.extend(reindexed_refl)
-        logger.info(
-            "Saving reindexed experiments to %s", self._params.output.experiments
+        else:
+            d_min = params.d_min
+        merged_reflections = prepare_merged_reflection_table(
+            experiments, reflection_tables, d_min
+        )
+
+        run_systematic_absences_checks(
+            experiments,
+            merged_reflections,
+            float(params.systematic_absences.significance_level),
         )
-        reindexed_experiments.as_file(self._params.output.experiments)
+
+    logger.info("Saving reindexed experiments to %s", params.output.experiments)
+    experiments.as_file(params.output.experiments)
+    if params.laue_group is not None:
         logger.info(
             "Saving %s reindexed reflections to %s",
-            len(reindexed_reflections),
-            self._params.output.reflections,
+            len(reflection_tables[0]),
+            params.output.reflections,
+        )
+        reflection_tables[0].as_file(params.output.reflections)
```

Comment:
Is it a potential cause for confusion that a reflection file is only exported `if params.laue_group is not None`? Since in a lot of cases the output reflections will be identical to the input file anyway if the cb_op is identity, would it not be simplest just to always re-export the reflections? We could always check if the `params.output.reflections is not None` to allow the user to choose not to re-export the reflections.

Reply:
Yes I think that's a good idea to keep a consistent behaviour between options, I'll update